library/src/main/java/org/xmtp/android/library/Conversations.kt

```diff
@@ -320,6 +302,106 @@ data class Conversations(
         return conversation
     }
 
+    suspend fun listGroups(
+        after: Date? = null,
+        before: Date? = null,
+        limit: Int? = null,
+    ): List<Group> {
+        val ffiGroups = libXMTPConversations?.listGroups(
+            opts = FfiListConversationsOptions(
+                after?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                before?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                limit?.toLong()
+            )
+        ) ?: throw XMTPException("Client does not support V3 dms")
+
+        return ffiGroups.map {
+            Group(client, it)
+        }
+    }
+
+    suspend fun listDms(
+        after: Date? = null,
+        before: Date? = null,
+        limit: Int? = null,
+    ): List<Dm> {
+        if (client.hasV2Client) throw XMTPException("Only supported for V3 only clients.")
+        val ffiDms = libXMTPConversations?.listDms(
+            opts = FfiListConversationsOptions(
+                after?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                before?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                limit?.toLong()
+            )
+        ) ?: throw XMTPException("Client does not support V3 dms")
+
+        return ffiDms.map {
+            Dm(client, it)
+        }
+    }
+
+    suspend fun listConversations(
+        after: Date? = null,
+        before: Date? = null,
+        limit: Int? = null,
+        order: ConversationOrder = ConversationOrder.CREATED_AT,
+        consentState: ConsentState? = null,
+    ): List<Conversation> {
+        if (client.hasV2Client)
+            throw XMTPException("Only supported for V3 only clients.")
+
+        val ffiConversations = libXMTPConversations?.list(
+            FfiListConversationsOptions(
+                after?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                before?.time?.nanoseconds?.toLong(DurationUnit.NANOSECONDS),
+                limit?.toLong()
+            )
+        ) ?: throw XMTPException("Client does not support V3 dms")
+
+        val filteredConversations = filterByConsentState(ffiConversations, consentState)
+        val sortedConversations = sortConversations(filteredConversations, order)
+
+        return filteredConversations.map { it.toConversation() }
+    }
+
+    private fun sortConversations(
+        conversations: List<FfiConversation>,
+        order: ConversationOrder,
+    ): List<FfiConversation> {
+        return when (order) {
+            ConversationOrder.LAST_MESSAGE -> {
+                conversations.map { conversation ->
+                    val message =
+                        conversation.findMessages(FfiListMessagesOptions(null, null, null, null))
```

Comment:
I want to limit and order these descending

https://github.com/xmtp/libxmtp/pull/1164